### ISSUE TITLE
"Scanning source files" is slow in IntelliJ

### ISSUE THREAD

**Issue by vjpr** [wallabyjs/public#1577]
### Issue description or question

Has anything changed in how this works, or is there a way to debug this?

I work in a lerna monorepo with lots of nested `node_modules` and the "Scanning source files" step has become very slow.

I have tried a bunch of different patterns and it doesn't seem to make sense.

Is there a way to see which files are being scanned?

From https://github.com/wallabyjs/public/issues/605#issuecomment-220875636:

> make the initial scan smarter by determining the list of roots to scan from the files list. I'll look into this now.

How are the roots determined?

### Wallaby.js configuration file

Can you see anything wrong with this? There are 5 packages inside this dir, and they have very few source files but it is taking 30s to startup. Feels like it must be looking through `node_modules`. My `node_modules` are all excluded roots in the IntelliJ project too.

```javascript
{ files: 
   [ { pattern: 'packages/public/babel/.babelrc',
       instrument: false,
       load: false },
     'packages/public/babel/**/*.json',
     'packages/public/babel/**/*.js',
     'packages/public/babel/**/*.proto',
     '!packages/public/babel/**/*.test.js',
     '!packages/public/babel/**/node_modules/**',
     '!packages/public/babel/**/flow-typed/**',
     '!packages/public/babel/**/dist/**',
     '!packages/public/babel/**/build/**',
     '!packages/public/babel/**/.next/**',
     '!packages/public/babel/**/.cache/**',
     '!packages/public/babel/**/.dev/**',
     '!packages/public/babel/**/*.flow.js' ],
  tests: 
   [ 'packages/public/babel/**/*.test.js',
     '!packages/public/babel/**/node_modules/**',
     '!packages/public/babel/**/flow-typed/**',
     '!packages/public/babel/**/dist/**',
     '!packages/public/babel/**/build/**',
     '!packages/public/babel/**/.next/**',
     '!packages/public/babel/**/.cache/**',
     '!packages/public/babel/**/.dev/**',
     '!packages/public/babel/**/*.flow.js' ],
  babelCompileGlobs: 
   [ 'packages/public/babel/**/*.js',
     'packages/public/babel/**/.*/*.js' ] }
```

### Code editor or IDE name and version

```
IntelliJ IDEA 2018.1 (Ultimate Edition)
Build #IU-181.4203.519, built on March 21, 2018
Licensed to Vaughan Rouesnel
Subscription is active until April 3, 2018
JRE: 1.8.0_152-release-1136-b20 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.12.6
```

### OS name and version

Windows
**OSX**
Linux

**Comment by vjpr** [comment]
After a little digging, maybe its perf related to `String.split`:

![screen shot 2018-03-23 at 6 11 24 pm](https://user-images.githubusercontent.com/281413/37843711-01e3e182-2ec6-11e8-86d7-debcb7ab60aa.png)

![screen shot 2018-03-23 at 6 12 53 pm](https://user-images.githubusercontent.com/281413/37843708-0016fc40-2ec6-11e8-922a-b77c30e57008.png)

![screen shot 2018-03-23 at 6 14 01 pm](https://user-images.githubusercontent.com/281413/37843703-fe20f346-2ec5-11e8-9c7f-3c11d1c4e384.png)

**Comment by ArtemGovorov** [comment]
Thanks for the detailed report, the issue is fixed and the fix is published in 1.0.111.

> I wonder if there is any loggers that you could add to be able to debug these issues in future.

Yes, you may do `Wallaby:trace`, but the scanned files were not logged anyway.
